activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -266,10 +266,11 @@ def supports_primary_key? #:nodoc:
 
       # Enable standard-conforming strings if available.
       def set_standard_conforming_strings
-        old, self.client_min_messages = client_min_messages, 'panic'
-        execute('SET standard_conforming_strings = on') rescue nil
-      ensure
-        self.client_min_messages = old
+        execute(<<-SQL, 'SCHEMA')
+          UPDATE pg_settings
+          SET setting = 'on'
+          WHERE name = 'standard_conforming_strings'
+        SQL
```

Comment:
Let's make sure to backport the most modern version of this: https://github.com/rails/rails/commit/6a9323bca4a610cfb8ba3db5cf8ec1dbe762703e